trussworks/react-uswds
# feat!: Webpack/Jest to Vite/Vitest migration

# Summary

This is a conversion from using webpack/jest as bundler/test library to vite/vitest. This will ensure the react-uswds library can support the new ESM standard, while maintaining backwards compatibility with CJS. All npm scripts have been updated accordingly to perform the same intended actions. Building is two steps: build the library and build the uswds assets (css, related files, etc.) dictated by the `--mode uswds` argument (running just `build` will do these steps for you).

Vite
- Output bundles for all formats Vite supports (umd, cjs, iife, es)
- Output source maps for all bundles
- package.json updated with (hopefully) best practices so that the correct bundle will always be selected by users of this library
- Parity with all other outputs (styles and assets)

Vitest
- Maintains parity with jest configuration (coverage, etc.)

Also included:
- new eslinting for storybook and vitest with updates to fix warnings/errors they detected
- husky added to project, with the precommit script set to run checks on: lint, format, test related (no more forgetting to run that darn format:fix!)
- all dependencies up-to-date as of 2/4/24

Notes:
- happo unfortunately still requires webpack and babel (if you ever move off of happo, these dependencies can likely be dropped)
- the happo and happo-ci scripts were updated to properly point to the renamed `.happo.cjs` file
- A new `bundles.test.tsx` file was added that is excluded during normal testing and is ran after building to do basic importing and rendering tests on each bundle format

## Related Issues or PRs

#192 

## How To Test

- Using whichever means available from your local project's package manager of choice: have it use this vite build for the react-uswds dependency
- Verify your project continues to work as expected
